raster-foundry/raster-foundry
# Add additional fields to user profile

## Overview

This PR adds the following additional fields to user profile and supports setting email notifications to be sent to personal email (other than login email) on the frontend:
 - [x] First Name
 - [x] Last Name
 - [x] Phone Number
 - [x] Organization Name
 - [x] Organization Type (Commercial, Government, Non-Profit, Academic, and Military)
 - [x] Organization Website
 - [x] Profile Website
 - [x] Profile Bio
 - [x] Profile URL

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

Save personal info, which should be optional at this stage:
<img width="1473" alt="screen shot 2018-07-17 at 4 22 03 pm" src="https://user-images.githubusercontent.com/16109558/42843492-3cc3e3f6-89de-11e8-875d-3f1964d5911b.png">

Change email notification:
<img width="1473" alt="screen shot 2018-07-17 at 4 21 38 pm" src="https://user-images.githubusercontent.com/16109558/42843504-497e2304-89de-11e8-8aa3-bbc8ce70e824.png">


### Notes

We need to support sending emails to user's personal email address as well on backend and batch, if a users sets so on the frontend. I think the user should only be able to receive emails sent to one of the two email addresses.


## Testing Instructions

 * Run migrations
 * Go to your profile settings page and update your personal info. Check if it succeeds
 * Go to your profile email notification settings, if you have set a personal email address, it should appear here as well. Update your preferences on this page and check if it succeeds
 * These new fields should all be considered as private information, so check if you can list such info in other endpoints that list users

Closes https://github.com/azavea/raster-foundry-platform/issues/394
